tools/audioconv64/conv_xm64.c

```diff
@@ -77,10 +77,37 @@ int xm_convert(const char *infn, const char *outfn) {
 				memset(sout+length, 0, MIXER_LOOP_OVERREAD);
 				break;
 			case XM_FORWARD_LOOP:
-				sout = malloc(loop_end + MIXER_LOOP_OVERREAD);
-				length = loop_end;
-				memcpy(sout, s->data8, loop_end);
-				memmove(sout + loop_end, s->data8 + loop_end - loop_length, MIXER_LOOP_OVERREAD);
+				// Special case for odd-sized loops of 8-bit samples. We cannot
+				// properly handle these at runtime because they cannot be DMA'd
+				// as they change the 2-byte phase between ROM and RAM. 
+				// xm64.c will decrease the loop length by 1 byte to playback them,
+				// but this can affect the period in case of short loops:
+				// for instance, a 13-bytes loop shortened 12-bytes change the
+				// period by 7%, which can be several notes of difference at
+				// high frequencies.
+				// So for short loops (<1000 bytes), we just duplicate the loop
+				// itself to make it of even size. For longer loops, the period
+				// error made by xm64 when shortening is < 0.1%, which isn't
+				// audible.
+				if (bps == 1 && loop_length%2 == 1 && loop_length < 1024) {
```

Comment:
Defining the loop length threshold as a constant would be better I think. Maybe even go as far as making it configurable via CLI argument (If there is any benefit to that)?

Reply:
Done. I don't think it makes sense to expose it as a command line flag; it's a rather obscure implementation detail that is hard to explain and tuning it has very limited effect on file size.